src/gui/folderwatcher.cpp

```diff
@@ -190,4 +209,74 @@ void FolderWatcher::changeDetected(const QStringList &paths)
     }
 }
 
+QString FolderWatcher::possiblyAddUnlockedFilePath(const QString &path)
+{
+    qCInfo(lcFolderWatcher) << "Checking if it is a lock file:" << path;
+    const auto pathSplit = path.split(QLatin1Char('/'), Qt::SkipEmptyParts);
+    if (pathSplit.isEmpty()) {
+        return {};
+    }
+    QString lockFilePatternFound;
+    for (const auto &lockFilePattern : lockFilePatterns) {
+        if (pathSplit.last().startsWith(lockFilePattern)) {
+            lockFilePatternFound = lockFilePattern;
+            break;
+        }
+    }
+
+    if (lockFilePatternFound.isEmpty() || QFileInfo::exists(path)) {
+        return {};
+    }
+
+    qCInfo(lcFolderWatcher) << "Found a lock file with prefix:" << lockFilePatternFound << "in path:" << path;
+
+    const auto lockFilePathWitoutPrefix = QString(path).replace(lockFilePatternFound, QStringLiteral(""));
+    auto lockFilePathWithoutPrefixSplit = lockFilePathWitoutPrefix.split(QLatin1Char('.'));
+
+    if (lockFilePathWithoutPrefixSplit.size() < 2) {
+        return {};
+    }
+
+    auto extensionSanitized = lockFilePathWithoutPrefixSplit.takeLast().toStdString();
+    // remove possible non-alphabetical characters at the end of the extension
+    extensionSanitized.erase(
+        std::remove_if(extensionSanitized.begin(), extensionSanitized.end(), [](const auto &ch) {
```

Comment:
```suggestion
        std::remove_if(extensionSanitized.begin(), extensionSanitized.end(), [] (const auto &ch) {
```

Reply:
@mgallien I disagree. This is rather an unusual format. Most of our code does not have that space as well as examples you find online. Is this a recommendation from the latest standard? Otherwise, I would keep it without the space